runtime/syntax/qml.vim

```diff
@@ -0,0 +1,1130 @@
+syntax keyword qmlObjectLiteralType WaylandQuickItem
+syntax keyword qmlObjectLiteralType WaylandSeat
+syntax keyword qmlObjectLiteralType WaylandSurface
+syntax keyword qmlObjectLiteralType WaylandView
+syntax keyword qmlObjectLiteralType Waypoint
+syntax keyword qmlObjectLiteralType WebChannel
+syntax keyword qmlObjectLiteralType WebEngine
+syntax keyword qmlObjectLiteralType WebEngineAction
+syntax keyword qmlObjectLiteralType WebEngineCertificateError
+syntax keyword qmlObjectLiteralType WebEngineClientCertificateOption
+syntax keyword qmlObjectLiteralType WebEngineClientCertificateSelection
+syntax keyword qmlObjectLiteralType WebEngineDownloadItem
+syntax keyword qmlObjectLiteralType WebEngineHistory
+syntax keyword qmlObjectLiteralType WebEngineHistoryListModel
+syntax keyword qmlObjectLiteralType WebEngineLoadRequest
+syntax keyword qmlObjectLiteralType WebEngineNavigationRequest
+syntax keyword qmlObjectLiteralType WebEngineNewViewRequest
+syntax keyword qmlObjectLiteralType WebEngineNotification
+syntax keyword qmlObjectLiteralType WebEngineProfile
+syntax keyword qmlObjectLiteralType WebEngineScript
+syntax keyword qmlObjectLiteralType WebEngineSettings
+syntax keyword qmlObjectLiteralType WebEngineView
+syntax keyword qmlObjectLiteralType WebSocket
+syntax keyword qmlObjectLiteralType WebSocketServer
+syntax keyword qmlObjectLiteralType WebView
+syntax keyword qmlObjectLiteralType WebViewLoadRequest
+syntax keyword qmlObjectLiteralType WeekNumberColumn
+syntax keyword qmlObjectLiteralType WheelEvent
+syntax keyword qmlObjectLiteralType WheelHandler
+syntax keyword qmlObjectLiteralType Window
+syntax keyword qmlObjectLiteralType WlScaler
+syntax keyword qmlObjectLiteralType WlShell
+syntax keyword qmlObjectLiteralType WlShellSurface
+syntax keyword qmlObjectLiteralType WorkerScript
+
+syntax keyword qmlObjectLiteralType XAnimator
+syntax keyword qmlObjectLiteralType XdgDecorationManagerV1
+syntax keyword qmlObjectLiteralType XdgOutputManagerV1
+syntax keyword qmlObjectLiteralType XdgPopup
+syntax keyword qmlObjectLiteralType XdgPopupV5
+syntax keyword qmlObjectLiteralType XdgPopupV6
+syntax keyword qmlObjectLiteralType XdgShell
+syntax keyword qmlObjectLiteralType XdgShellV5
+syntax keyword qmlObjectLiteralType XdgShellV6
+syntax keyword qmlObjectLiteralType XdgSurface
+syntax keyword qmlObjectLiteralType XdgSurfaceV5
+syntax keyword qmlObjectLiteralType XdgSurfaceV6
+syntax keyword qmlObjectLiteralType XdgToplevel
+syntax keyword qmlObjectLiteralType XdgToplevelV6
+syntax keyword qmlObjectLiteralType XmlListModel
+syntax keyword qmlObjectLiteralType XmlRole
+syntax keyword qmlObjectLiteralType XYPoint
+syntax keyword qmlObjectLiteralType XYSeries
+
+syntax keyword qmlObjectLiteralType YAnimator
+
+syntax keyword qmlObjectLiteralType ZoomBlur
+
+" }}}
+
+if get(g:, 'qml_fold', 0)
+  syn match   qmlFunction      "\<function\>"
+  syn region  qmlFunctionFold  start="\<function\>.*[^};]$" end="^\z1}.*$" transparent fold keepend
```

Comment:
This doesn't look right.  `\z1` would need a corresponding `\z(\)` group in the start pattern.

It looks like the whitespace (indent) is supposed to match.  Try this

```vim
syn region  qmlFunctionFold  start="^\z(\s*\)\<function\>.*[^};]$" end="^\z1}.*$" transparent fold keepend
```